pkg/repository/ensurer.go

```diff
@@ -130,14 +136,14 @@ func (r *RepositoryEnsurer) EnsureRepo(ctx context.Context, namespace, volumeNam
 
 	log.Debug("Acquired lock")
 
-	selector := labels.SelectorFromSet(repoLabels(volumeNamespace, backupLocation))
+	selector := labels.SelectorFromSet(repoLabels(volumeNamespace, backupLocation, repositoryType))
 
 	repos, err := r.repoLister.BackupRepositories(namespace).List(selector)
 	if err != nil {
 		return nil, errors.WithStack(err)
 	}
 	if len(repos) > 1 {
-		return nil, errors.Errorf("more than one ResticRepository found for workload namespace %q, backup storage location %q", volumeNamespace, backupLocation)
+		return nil, errors.Errorf("more than one ResticRepository found for workload namespace %q, backup storage location %q, repository type %q", volumeNamespace, backupLocation, repositoryType)
```

Comment:
ResticRepository should be BackupRepository

Reply:
Done